app/services/mapping/declaration/AgentBuilder.scala

```diff
@@ -31,21 +29,13 @@ class AgentBuilder @Inject()() extends ModifyingBuilder[ExportsCacheModel, Decla
 
   override def buildThenAdd(exportsCacheModel: ExportsCacheModel, declaration: Declaration): Unit =
     exportsCacheModel.parties.representativeDetails.foreach { representativeDetails =>
-      if (RepresentativeDetails.isDefined(representativeDetails)) {
+      if (isDefined(representativeDetails)) {
         declaration.setAgent(createAgent(representativeDetails))
       }
     }
 
-}
-
-object AgentBuilder {
-
-  def build(implicit cacheMap: CacheMap): Declaration.Agent =
-    cacheMap
-      .getEntry[RepresentativeDetails](RepresentativeDetails.formId)
-      .filter(RepresentativeDetails.isDefined)
-      .map(data => createAgent(data))
-      .orNull
+  def isDefined(representativeDetails: RepresentativeDetails): Boolean =
+    representativeDetails.details.isDefined && (representativeDetails.details.get.eori.isDefined || representativeDetails.details.get.address.isDefined)
```

Comment:
here we could write this with combinator on `Option` without using `get` like this:
```
representativeDetails.details.exist(details => details.eori.isDefined && details.address.isDefined)
```

Reply:
We could (and will).  In my defence, I moved this code from elsewhere.